prototypes/basic/app/views/review.html

```diff
@@ -2,14 +2,67 @@
 
 {% block pageTitle %} {{ serviceName }} – GOV.UK Prototype Kit {% endblock %}
 {% block beforeContent %}
-{{ govukBackLink({ text: "Back", href: "javascript:window.history.back()" }) }}
+{{
+    govukBreadcrumbs({
+      items: [
+        {
+          text: "Home",
+          href: "/"
+        },
+        {
+            text: "Upload spreadsheet",
+            href: "/upload"
+        },
+        {
+          text: "Select sheet",
+          href: "/select_sheet"
+        },
+        {
+            text: "Select header row",
+            href: "/select_header_row"
+        },
+        {
+            text: "Select footer row",
+            href: "/select_footer_row"
+        },
+        {
+            text: "Identify columns",
+            href: "/mapping"
+        },
+        {
+            text: "Review data",
+            href: "/review"
+        }
+      ]
+    })
+  }}
 {% endblock %}
 
 {% block content %}
+{% set results = importerMappedData(data) %}
+
 <div class="govuk-grid-row">
     <div class="govuk-grid-column-two-thirds">
         <h1 class="govuk-heading-l">Review your data</h1>
 
+        <p class="govuk-body">
+            During import there were <strong>{{results.errorCount}}</strong> errors and <strong>{{results.warningCount}}</strong> warnings.
+        </p>
+
+        <p class="govuk-body">
+        The uploaded data contains the following data:
+
+            <ul class="govuk-list govuk-list--bullet">
+                <li><strong>{{ results.totalCount }}</strong> rows of data</li>
+                <li>a total salary commitment from the 'Salary' column of <strong>{{ data_sum(data, 'Salary') | currency }}</strong>.</li>
+                <li>an average salary from the 'Salary' column of <strong>{{ data_avg(data, 'Salary') | currency }}</strong>.</li>
```

Comment:
Do these handle decimals correctly? I've uploaded a CSV with values like `3312.45` and it has summed values into the millions and averaged the values into the hundred-thousands?

Reply:
Ah no it's not proper parsing but will fix this as a special case.

Reply:
Improved number parsing in 307045c

It might be worth us investing some time later on in a decent type parser to accurarely identify types given a bytestream and a set of options (e.g. how strict the parsing should be) as it would be useful both for this and in type validation as well as other parts of the future system.